src/Datadog.Trace.ClrProfiler.Managed/Emit/ModuleLookup.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Emit
+{
+    internal static class ModuleLookup
+    {
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(ModuleLookup));
+
+        private static object _populationLock = new object();
+        private static ConcurrentDictionary<Guid, Module> _modules = new ConcurrentDictionary<Guid, Module>();
+
+        public static Module Get(Guid moduleVersionId)
+        {
+            Module value = null;
+
+            // Mimics a wait if the population code is running
+            Monitor.Enter(_populationLock);
+            Monitor.Exit(_populationLock);
```

Comment:
I don't like this, any better ideas?

Reply:
Not sure what you're intent is here, but you might want to look into other [thread synchronization primitives](https://docs.microsoft.com/en-us/dotnet/standard/threading/overview-of-synchronization-primitives), like an `ManualResetEventSlim` or a `Mutex`.